introspection-engine/connectors/sql-introspection-connector/tests/test_harness/test_api.rs

```diff
@@ -262,3 +262,75 @@ pub async fn sqlite_test_api(db_name: &'static str) -> TestApi {
         introspection_connector,
     }
 }
+
+pub async fn mssql_2017_test_api(schema: &'static str) -> TestApi {
+    mssql_test_api(mssql_2017_url("master"), schema).await
+}
+
+pub async fn mssql_2019_test_api(schema: &'static str) -> TestApi {
+    mssql_test_api(mssql_2019_url("master"), schema).await
+}
+
+pub async fn mssql_test_api(connection_string: String, schema: &'static str) -> TestApi {
+    let connection_string = format!("{};schema={}", connection_string, schema);
+    let database = Quaint::new(&connection_string).await.unwrap();
+    let connection_info = database.connection_info().to_owned();
+
+    // Mickie misses DROP SCHEMA .. CASCADE, so what we need to do here is to
+    // delete first the foreign keys, then all the tables from the test schema
+    // to allow a clean slate for the next test.
+
+    let drop_fks = format!(
+        r#"
+        DECLARE @stmt NVARCHAR(max)
+        DECLARE @n CHAR(1)
+
+        SET @n = CHAR(10)
+
+        SELECT @stmt = ISNULL(@stmt + @n, '') +
+            'ALTER TABLE [' + SCHEMA_NAME(schema_id) + '].[' + OBJECT_NAME(parent_object_id) + '] DROP CONSTRAINT [' + name + ']'
+        FROM sys.foreign_keys
+        WHERE SCHEMA_NAME(schema_id) = '{0}'
+
+        EXEC SP_EXECUTESQL @stmt
+        "#,
+        schema
+    );
+
+    let drop_tables = format!(
+        r#"
+        DECLARE @stmt NVARCHAR(max)
+        DECLARE @n CHAR(1)
+
+        SET @n = CHAR(10)
+
+        SELECT @stmt = ISNULL(@stmt + @n, '') +
+            'DROP TABLE [' + SCHEMA_NAME(schema_id) + '].[' + name + ']'
+        FROM sys.tables
+        WHERE SCHEMA_NAME(schema_id) = '{0}'
+
+        EXEC SP_EXECUTESQL @stmt
+        "#,
+        schema
+    );
```

Comment:
I don't remember if it's exactly the same, but if this code is duplicated between this and the migration engine, would it make sense to put it in the `test-setup` crate?

Reply:
Fixed in https://github.com/prisma/prisma-engines/pull/1186/commits/9b5fc892eaa4e1cd69516066db353618250cfb3c